### DISCUSSION TITLE
Pandas and Python 3.9 issue

### DISCUSSION THREAD

**Discussion by wkarney** [wkarney/minted-export#4] [Q&A] [answered]
**Questions or Comments**

Let me know if you are encountering any difficulties with making the scripts work on your machine here and I can help troubleshoot!

**Comment by lenoxbooks** [comment]
Hey, thanks for putting this together. Any ideas on what could be causing this:

`Traceback (most recent call last):
  File "minted_api_request.py", line 47, in <module>
    address_book = pd.DataFrame(listings)
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py", line 468, in __init__
    mgr = init_dict(data, index, columns, dtype=dtype)
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/core/internals/construction.py", line 283, in init_dict
    return arrays_to_mgr(arrays, data_names, index, columns, dtype=dtype)
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/core/internals/construction.py", line 78, in arrays_to_mgr
    index = extract_index(arrays)
  File "/opt/anaconda3/lib/python3.7/site-packages/pandas/core/internals/construction.py", line 387, in extract_index
    raise ValueError("If using all scalar values, you must pass an index")
ValueError: If using all scalar values, you must pass an index`

**Comment by wkarney** [comment]
I think your username and password were not entered correctly.  If the correct credentials are not entered then the response will return: `{'detail': 'Authentication credentials were not provided.'}`. And Pandas will raise an exception when trying to convert that into a data frame in line 47.

Let me know if this fixes your issue @lenoxbooks and thanks for using my script!

**Comment by John5373** [comment, marked as answer]
Thanks for your help with this. I am having an issue installing the Panda dependency. Running Python 3.9. Here is my error: 

> Failed to build pandas
> ERROR: Could not build wheels for pandas which use PEP 517 and cannot be installed directly

EDIT: Got it working--just had to manually install all of the dependencies. HUGE thanks for putting this together!

**Reply by bimrin** [reply]
@maniscalcojohn can you indicate which dependencies you installed?  Bit of a novice and am running into the same issue.

EDIT: Actually nevermind I figured out my issue.  After manually installing all dependencies, I kept running the requirements step instead of just running the script at that point.  I did run into an issue (it might be specific to my config, that I had to create the data directory, it wouldn't create for me)

**Reply by wkarney** [reply]
Glad you figured out a work around, @bimrin. I'll look into doing some refactoring and testing to make sure people don't run into issues...clearly something with Python 3.9 and package installations is being annoying (Python 3.9 wasn't released when I had developed this...)

If you don't mind me asking: What's your particular setup/config? macOS/Windows/Linux? Straight python install or through Anaconda or miniconda?

**Reply by bimrin** [reply]
> Glad you figured out a work around, @bimrin. I'll look into doing some refactoring and testing to make sure people don't run into issues...clearly something with Python 3.9 and package installations is being annoying (Python 3.9 wasn't released when I had developed this...)
> 
> If you don't mind me asking: What's your particular setup/config? macOS/Windows/Linux? Straight python install or through Anaconda or miniconda?

I tried first in Windows and then didn't want to bother trying to get that configuration all working, so I switched over to MacOS where I could just brew up python, and get the virtual environment going.  (Windows was giving me issues with the Virtual Env)

**Reply by wkarney** [reply]
Yeah, getting that config right with Windows can be a pain. macOS (or Linux) definitely seem to be the easiest/fastest way to get things running.

**Comment by wkarney** [comment]
Glad you got it working and glad it helped! @maniscalcojohn
